//   Copyright 2023 Dapta LTD
//
//   Licensed under the Apache License, Version 2.0 (the "License");
//   you may not use this file except in compliance with the License.
//   You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the License is distributed on an "AS IS" BASIS,
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and
//   limitations under the License.

syntax = "proto3";
import "google/protobuf/struct.proto";

message ComponentRequest{
    google.protobuf.Struct inputs = 1;
}

message ComponentResponse{
    string outputmsg = 1;
    google.protobuf.Struct outputs = 2; 
}

message ComponentLargeMessage{
    string jsonstr = 1;
}

service Component {
    rpc Setup (ComponentRequest) returns (ComponentResponse);
    rpc Compute (ComponentRequest) returns (ComponentResponse);
    rpc LargeCompute (stream ComponentLargeMessage) returns (stream ComponentLargeMessage);
}